### PR TITLE
Refactor itests to allow adding multiple transport sinks

### PR DESCRIPTION
**Pull request by nehalrp**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,15 +28,9 @@ test: vendor generate vet
 	@echo "Executing tests ..."
 	go test -race -v ${GO_TEST_EXTRAS} ./...
 
-itests-circleci:
```
Comment (stlava) [inline Makefile]: Need to update `Makefile` after removing this.

File: itests/docker-compose.yml [hunk 2]
```diff
@@ -10,16 +10,16 @@ services:
     networks:
       - net
```
Comment (stlava) [inline itests/docker-compose.yml]: After you're done can you please run this file through a linter to fix all the different types of spacings

File: itests/itests_runner.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    else
+        _base_testfiles=$(cd tests/base && ls -d */ | sed 's#/##' | sed -e 's/^/base\//')
+        _specific_testfiles=$(cd tests/${_transport_sink} && ls -d */ | sed 's#/##' | sed -e "s/^/${_transport_sink}\//")
+    fi
+
+    echo "${_base_testfiles} ${_specific_testfiles}"
+}
+
+# Kinesis
+set -a ; . contexts/kinesis.env ; set +a
```
Comment (stlava) [inline itests/itests_runner.sh]: How will this work with different contexts?

Comment (nehalrp) [reply itests/itests_runner.sh]: Added unsetting based on environment file.

**Pull request merged**